### PR TITLE
Improve download progress

### PR DESCRIPTION
**Pull request by nodakai**

### CODE REVIEW

File: src/rustup-cli/download_tracker.rs [hunk 1]
```diff
@@ -121,37 +121,31 @@ impl DownloadTracker {
     }
     /// Display the tracked download information to the terminal.
     fn display(&mut self) {
-        let total_h = HumanReadable(self.total_downloaded as u64);
+        let total_h = HumanReadable(self.total_downloaded as f64);
         let sum = self.downloaded_last_few_secs
                       .iter()
-                      .fold(0usize, |a, &v| a + v);
+                      .fold(0., |a, &v| a + v as f64);
```
Comment (tamird) [inline src/rustup-cli/download_tracker.rs]: what's the point of making this a float? this is simple addition of integers, there's no rounding here.

File: src/rustup-cli/download_tracker.rs [hunk 2]
```diff
@@ -121,37 +121,31 @@ impl DownloadTracker {
...
         let len = self.downloaded_last_few_secs.len();
         let speed = if len > 0 {
-            (sum / len) as u64
+            sum / len as f64
```
Comment (tamird) [inline src/rustup-cli/download_tracker.rs]: for consistency with the below i think this should be `sum as f64 / len`

Comment (nodakai) [inline src/rustup-cli/download_tracker.rs]: f64 / usize doesn't compile

Comment (tamird) [inline src/rustup-cli/download_tracker.rs]: then i guess you want `sum as f64 / len as f64`, right?

File: src/rustup-cli/download_tracker.rs (hunk 1 above)
Comment (nodakai) [inline src/rustup-cli/download_tracker.rs]: Hmm, this is just a matter of taste. I just wanted `sum` to be a float and I tend to apply int-to-float casting as early as possible when I know the end result is a float.

File: src/rustup-cli/download_tracker.rs (hunk 2 above)
Comment (nodakai) [inline src/rustup-cli/download_tracker.rs]: In a sense, yes. But in my version, `sum` is already of type `f64`.

File: src/rustup-cli/download_tracker.rs [hunk 3]
```diff
@@ -168,20 +162,36 @@ impl DownloadTracker {
...
 impl fmt::Display for HumanReadable {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        const KIB: f64 = 1024.0;
-        const MIB: f64 = 1048576.0;
-        let size = self.0 as f64;
+        if f.alternate() {  // repurposing the alternate mode for ETA
+            let sec = self.0;
+
+            if sec <= 0. {
+                write!(f, "Unknown")
```
Comment (nodakai) [inline src/rustup-cli/download_tracker.rs]: Oops, the condition to print "Unknown" is now wrong...

**Comment by brson** [comment]
Thanks for the improvements @nodakai, and for explaining the float math.

**Comment by nodakai** [comment]
@brson Sorry I need fix the condition to display "unknown". But haven't got the time... I'll prepare another patch.

**Pull request merged**